src/poetry/core/packages/dependency.py

```diff
@@ -260,7 +260,10 @@ def base_pep_508_name(self) -> str:
 
         constraint = self.constraint
         if isinstance(constraint, VersionUnion):
-            if constraint.excludes_single_version():
+            if (
+                constraint.excludes_single_version()
+                or constraint.excludes_single_wildcard_range()
```

Comment:
It seems that tests succeed if I remove this line. This should be tested, too.

Reply:
This whole block is more a short-circuit, even removing the if condition will pass the test suite. I will explicity add tests for version range exclusion, but it won't fail even if we remove this block.